redash/authentication/__init__.py

```diff
@@ -216,7 +217,8 @@ def log_user_logged_in(app, user):
 
 @login_manager.unauthorized_handler
 def redirect_to_login():
-    if request.is_xhr or "/api/" in request.path:
+    is_xhr = request.headers.get("X-Requested-With") == "XMLHttpRequest"
+    if is_xhr or "/api/" in request.path:
         response = jsonify(
```

Comment:
Not necessary to fix/change this here but fwiw jsonifying dictionary responses isn't required with Flask 2.0+